eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -56,6 +56,10 @@ class MultiPartHandler(nodeParams: NodeParams, register: ActorRef, db: IncomingP
       Try {
         val paymentPreimage = paymentPreimage_opt.getOrElse(randomBytes32)
         val paymentHash = Crypto.sha256(paymentPreimage)
+        // We must set a `min_final_cltv_expiry` bigger than our `fulfill_safety_before_timeout`. Otherwise when we receive
+        // a payment, if a new block is produced while we're waiting for our peer to acknowledge our fulfill we'll end up
+        // closing the channel because it violates `fulfill_safety_before_timeout`.
+        val minFinalExpiryDelta = nodeParams.fulfillSafetyBeforeTimeoutBlocks + 3
```

Comment:
How did you come up with the value `3`? Is it that pre d5ec6a56e842b8daabdc2dd91d43b238db956811, we had `fulfill-safety-before-timeout-blocks = 6` with `minFinalCltvExpiryDelta= 9` and `9-6=3`?

Reply:
I chose it completely arbitrarily :grinning: 
We need to add at least 1 to handle a new block being produced, and I figured maybe it's worth giving a bit more leeway than 1 block...
Do you think I should make this configurable? I could make `min-final-expiry-delta` a config value and verify that it's `> fulfill-safety-before-timeout`.

Reply:
Or maybe `+ 1` is good enough...

Reply:
> I could make min-final-expiry-delta a config value and verify that it's > fulfill-safety-before-timeout.

That seems sensible. We will need to introduce some hierarchy in the `nodeParams` soon.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1483/commits/01678eeadf94f0a9f9d5a9aee2dc42bd38d730aa, it feels cleaner.